bigquery/query_builder.py

```diff
@@ -1,9 +1,36 @@
+import re
+
 from logging import getLogger, NullHandler
 
+UNESCAPED_QUOTE = r"([^\\])'"
+
 logger = getLogger(__name__)
 logger.addHandler(NullHandler())
 
 
+class ParameterizedQuery:
+    def __init__(self, statement):
+        self._statement = statement
+
+    def bind(self, **kwargs):
+        escaped_args = self._escape_single_quotes(kwargs)
+        return self._statement.format(**escaped_args)
+
+    def _escape_single_quotes(self, arguments):
+        return {
+            arg_name: self._escape_string_value(arg_value)
+            for arg_name, arg_value in arguments.items()
+        }
+
+    def _escape_string_value(self, value):
+        try:
+            return re.sub(UNESCAPED_QUOTE, _replace_quote_with, value)
+        except TypeError:
```

Comment:
Are you sure that only `TypeError` can be raised here? If another error/exception is raised, it may break the execution.

Would it be better to verify the type instead? Something like:

```python
def _escape_string_value(self, value):
    if type(value) == str:
        return re.sub(UNESCAPED_QUOTE, _replace_quote_with, value)
    else:
        return value
```

Reply:
Yes, the Type error is raised when `value` is not a string,  a byte sequence, nor Pattern object. That's why I chose the `try/except` approach here. I think the code gets shorter. The reasoning is this:

1. First try to substitute the pattern
1. If the pattern is not found, then we return the same value
1. if the patter is found, then we return the _escaped_ value
1. if the substitution fails (it always will raise TypeError, since we're not doing weird stuff like passing flags to `re.sub()`

Reply:
Ok, it's fine. Since in the tests you are testing just the integer case... Do you think makes sense to verify other scenarios (different types)? So, that our tests are going to guarantee those types are handled in the `except`.

Reply:
I can add test agains `None`, `object()` and other collections, just to be sure :)